rotors_gazebo_plugins/src/gazebo_fw_dynamics_plugin.cpp

```diff
@@ -0,0 +1,406 @@
+           Eigen::Vector2d(0.0, aileron_sum)) +
+       aero_params_.c_lift_delta_flp.dot(
+           Eigen::Vector2d(0.0, flap_sum)));
+
+  Eigen::Vector3d forces_Wind(-drag, side_force, -lift);
+
+  // Non-dimensionalize the angular rates for inclusion in the computation of
+  // moments. To avoid division by zero, there is a minimum air speed threshold
+  // below which the values are zero.
+  double p_hat = (V < kMinAirSpeedThresh) ? 0.0 :
+      p * vehicle_params_.wing_span / (2.0 * V);
+  double q_hat = (V < kMinAirSpeedThresh) ? 0.0 :
+      q * vehicle_params_.chord_length / (2.0 * V);
+  double r_hat = (V < kMinAirSpeedThresh) ? 0.0 :
+      r * vehicle_params_.wing_span / (2.0 * V);
+
+  // Compute the moments in the wind frame.
+  double rolling_moment = q_bar_S * vehicle_params_.wing_span *
+      (aero_params_.c_roll_moment_beta.dot(
+           Eigen::Vector2d(0.0, beta)) +
+       aero_params_.c_roll_moment_p.dot(
+           Eigen::Vector2d(0.0, p_hat)) +
+       aero_params_.c_roll_moment_r.dot(
+           Eigen::Vector2d(0.0, r_hat)) +
+       aero_params_.c_roll_moment_delta_ail.dot(
+           Eigen::Vector2d(0.0, aileron_diff)) +
+       aero_params_.c_roll_moment_delta_flp.dot(
+           Eigen::Vector2d(0.0, flap_diff)));
+
+  double pitching_moment = q_bar_S * vehicle_params_.chord_length *
+      (aero_params_.c_pitch_moment_alpha.dot(
+           Eigen::Vector2d(1.0, alpha)) +
+       aero_params_.c_pitch_moment_q.dot(
+           Eigen::Vector2d(0.0, q_hat)) +
+       aero_params_.c_pitch_moment_delta_elv.dot(
+           Eigen::Vector2d(0.0, delta_elevator_)));
+
+  double yawing_moment = q_bar_S * vehicle_params_.wing_span *
+      (aero_params_.c_yaw_moment_beta.dot(
+           Eigen::Vector2d(0.0, beta)) +
+       aero_params_.c_yaw_moment_r.dot(
+           Eigen::Vector2d(0.0, r_hat)) +
+       aero_params_.c_yaw_moment_delta_rud.dot(
+           Eigen::Vector2d(0.0, delta_rudder_)));
+
+  Eigen::Vector3d moments_Wind(rolling_moment, pitching_moment, yawing_moment);
+
+  // Compute the thrust force in the body frame.
+  double thrust = aero_params_.c_thrust.dot(
+      Eigen::Vector3d(1.0, throttle_, throttle_ * throttle_));
+  double thrust_x = cos(vehicle_params_.thrust_inclination) * thrust;
+  double thrust_z = sin(vehicle_params_.thrust_inclination) * thrust;
+
+  Eigen::Vector3d force_thrust_B(thrust_x, 0.0, thrust_z);
+
+  // Compute the transform between the body frame and the wind frame.
+  double ca = cos(alpha);
+  double sa = sin(alpha);
+  double cb = cos(beta);
+  double sb = sin(beta);
+
+  Eigen::Matrix3d R_Wind_B;
+  R_Wind_B << ca * cb, sb, sa * cb,
+              -sb * ca, cb, -sa * sb,
+              -sa, 0.0, ca;
+
+  Eigen::Matrix3d R_Wind_B_t = R_Wind_B.transpose();
+
+  // Transform all the forces and moments into the body frame
+  Eigen::Vector3d forces_B = R_Wind_B_t * forces_Wind + force_thrust_B;
+  Eigen::Vector3d moments_B = R_Wind_B_t * moments_Wind;
```

Comment:
const ref?
Many of the variables (`const double ...`) above could also be const.